---
title: "Hierarchical Clustering on SCNA, Expression and Methylation"
author: "Shan Tang"
output: html_document
---

## Set Up
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(dplyr)
library(readxl)
library(ggplot2)
library(ComplexHeatmap)
library(dendextend)
library(circlize)
library(tibble)
```

## Legend Setting
```{r}
#Clinical outcome as top row annotation
ha_col1 = HeatmapAnnotation(df=data.frame(Type=sample_infor$type,
                DrugResp=sample_infor$dr,
                Metastasis=sample_infor$Metastasis,
                VitalStatus=sample_infor$Status),
    col = list(Type = c("Patient"="#56B4E9","Cell line"="#CC79A7"),
               DrugResp=c("Good responder" = "#009E73", "Poor responder"="#E69F00"),
               Metastasis= c("Non-metastatic"="#999999", "Metastatic"="#000000"),
               VitalStatus=c("Alive"="#999999", "Dead"="#000000") ),
    height = unit(1.25, "cm"), simple_anno_size_adjust = TRUE, 
    annotation_name_rot = 30)

# Chr annotation for CNV
chr<-data.frame(chr=c(1:22),n=rep(0,22))
temp<-1
for (i in 1:22) {
  chr[i,2]<-temp
  temp=temp+sum(cnv_order$chromosome_name==i)
  }
ha_chr = rowAnnotation(foo = anno_mark(at = chr$n, labels = as.character(c(1:22)), side="right",labels_gp = gpar(fontsize = 8) ))

## Average gene exp
MeanExpr = rowMeans(as.matrix(ge))
```


## SCNA
```{r SCNA}
ht_cnv= Heatmap(as.matrix(cnv_order), 
                #as.matrix(cnv30_order[,7:97]), 
        width = unit(10, "cm"), height = unit(7.5, "cm"),
        show_row_names = FALSE, show_column_names = FALSE, #column_names_gp = grid::gpar(fontsize = 6),
        name = "CopyNumber", 
        cluster_rows = FALSE, column_km=3,
        #column_split =ind$x,cluster_rows = FALSE, cluster_columns = FALSE, 
        #clustering_distance_columns="manhattan", clustering_method_columns = "ward.D",
        top_annotation = ha_col1,
        right_annotation = ha_chr
  )

ht1=draw(ht_cnv,heatmap_legend_side = "left",annotation_legend_side = "left") 
```

## Gene expression
```{r Expr}
col_uni<-colorRamp2(c(2.25, 9.5, 15), c("blue4", "white", "red3"))

ht_ge= Heatmap(as.matrix(ge),##heatmap1_ge_start
        col = col_uni,
        width = unit(10, "cm"), height = unit(7.5, "cm"),
        show_row_names = FALSE, show_column_names = FALSE, # column_names_gp = grid::gpar(fontsize = 6),
        name = "Expression", 
        column_km=3, 
        top_annotation = ha_col1
        ) + 
  Heatmap(as.matrix(hob), ##heatmap2_Human osteoblasts
          col = col_uni, 
          name = "HOB_Expr", width = unit(5, "mm"), column_names_gp = grid::gpar(fontsize = 8),
          show_row_names = FALSE, cluster_columns = FALSE,show_heatmap_legend = FALSE) +  ##heatmap2_hob_end
  Heatmap(MeanExpr,
          col = col_uni,
          name = "MeanExpr", width = unit(5, "mm"),column_names_gp = grid::gpar(fontsize = 8),
          show_row_names = FALSE,show_heatmap_legend = FALSE) ##heatmap3_base_end

ht2=draw(ht_ge,heatmap_legend_side = "left",annotation_legend_side = "left")
```

## Gene methylation
```{r Methy}
col_methy = colorRamp2(c(0, 0.5, 1), c("#0000FF", "white", "#FFFF00"))

ht_methy= Heatmap(as.matrix(methy), 
        col = col_methy, name = "Methylation", 
        width = unit(10, "cm"), height = unit(8, "cm"),
        show_row_names = FALSE, show_column_names = FALSE, 
        column_km=3,
        top_annotation = ha_col1,
        bottom_annotation = ha_group3
        )+
  Heatmap(as.matrix(methy_bone), name = "Bone",
          col = col_methy,
          show_column_dend = FALSE, show_heatmap_legend = FALSE, show_row_names = FALSE,
          width = unit(6, "mm"),column_names_gp = grid::gpar(fontsize = 8)
          )+
  Heatmap(methy_CpGs, name = "CpG island", 
          col = colorRamp2(c(0, 10), c("white", "black")), 
          show_row_names = FALSE, width = unit(3, "mm"),column_names_gp = grid::gpar(fontsize = 8)
          )
 
ht3=draw(ht_methy,heatmap_legend_side = "left",annotation_legend_side = "left") 
```

